src/test/java/org/springframework/integration/dsl/samples/filesplitter2aggregator/SplitFileAndAggregate.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.dsl.samples.filesplitter2aggregator;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.Aggregator;
+import org.springframework.integration.annotation.ReleaseStrategy;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.file.Files;
+import org.springframework.integration.dsl.support.Transformers;
+import org.springframework.integration.file.FileHeaders;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.xml.transformer.ResultToStringTransformer;
+import org.springframework.messaging.Message;
+import org.springframework.oxm.Marshaller;
+import org.springframework.oxm.jaxb.Jaxb2Marshaller;
+
+/**
+ * @author Artem Bilan
+ * @since 1.2
+ */
+@SpringBootApplication
+public class SplitFileAndAggregate {
+
+	public static void main(String[] args) throws Exception {
+		ConfigurableApplicationContext context = new SpringApplicationBuilder(SplitFileAndAggregate.class)
+				.web(false)
+				.run(args);
+		System.out.println("Put a file with a .txt extension in /tmp/in with names,"
+				+ "\nthe file will be read, names started with 'X' will be filtered and "
+				+ "the result will be stored in XML format in the "
+				+ "\n/tmp/out"
+				+ "\n\nHit enter to terminate");
+		System.in.read();
+		context.close();
+	}
+
+	@Bean
+	public Marshaller jaxbMarshaller() {
+		Jaxb2Marshaller jaxb2Marshaller = new Jaxb2Marshaller();
+		jaxb2Marshaller.setPackagesToScan(Names.class.getPackage().getName());
+		jaxb2Marshaller.setMarshallerProperties(Collections
+				.singletonMap(javax.xml.bind.Marshaller.JAXB_FORMATTED_OUTPUT, true));
+		return jaxb2Marshaller;
+	}
+
+	@Bean
+	public IntegrationFlow fileSplitterAggregator() {
+		return IntegrationFlows
+				.from(Files.inboundAdapter(new File("/tmp/in"))
+								.autoCreateDirectory(true)
+								.patternFilter("*.txt"),
+						e -> e.poller(Pollers.fixedDelay(5000)))
+				.split(Files.splitter()
+						.markers()
+						.applySequence(true))
+				.filter(p -> !(p instanceof FileSplitter.FileMarker),
+						e -> e.discardChannel("aggregatorChannel"))
```

Comment:
Yep! That's why I decided to make it as a public resource, instead of just an answer on SO